### PR TITLE
added a few java docs (Sentry, Hub and SentryClient)

### PR DESCRIPTION
**Pull request by marandaneto**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [X] Enhancement
- [ ] Refactoring

## :scroll: Description
added a few java docs.

## :bulb: Motivation and Context
we are poor of javadocs.

## :pencil: Checklist

- [X] I reviewed submitted code
- [ ] I added tests to verify changes
- [X] All tests passing

## :crystal_ball: Next steps
more and more javadocs

### CODE REVIEW

File: sentry-core/src/main/java/io/sentry/core/Sentry.java [hunk 1]
```diff
@@ -69,110 +93,224 @@ public static void init(@NotNull OptionsConfiguration<SentryOptions> optionsConf
     hub.close();
   }
 
+  /**
+   * Close the SDK
+   */
   public static synchronized void close() {
-    IHub hub = getCurrentHub();
+    IHub hub = getCurrentHub(); // TODO: should we not close all the hubs?
```
Comment (marandaneto) [inline sentry-core/src/main/java/io/sentry/core/Sentry.java]: @bruno-garcia

**Comment by bruno-garcia** [comment]
@marandaneto Ideally all public types their members. 

I'd say at least all public API from `Sentry`, `SentryOptions`, `SentryClient`, `Hub`, `SentryEvent` and `Scope`. Anything else major you'd suggest?

File: sentry-core/src/main/java/io/sentry/core/Sentry.java (hunk 1 above)
Comment (bruno-garcia) [reply sentry-core/src/main/java/io/sentry/core/Sentry.java]: This is a gap in the API. `Close` only closes the current Hub and this is the behavior in all SDKs AFAIK.
@HazAT was there any discussion over this issue?

We could hold weak references and dispatch close to all known hubs or something like that.
@mitsuhiko might have opinions here.

**Comment by marandaneto** [comment]
> CI is unhappy somehow

because of #224 already fixed it

File: sentry-core/src/main/java/io/sentry/core/Sentry.java (hunk 1 above)
Comment (HazAT) [reply sentry-core/src/main/java/io/sentry/core/Sentry.java]: I think it's fine just to close the current hub/client.
I would expect to call close doing it for all hubs.

**Pull request merged**